scipy/stats/stats.py

```diff
@@ -3692,6 +3695,163 @@ def f_oneway(*args, axis=0):
+        >>> alexandergovern(atlanta, chicago, houston, memphis)
+        AlexanderGovernResult(statistic=4.65087071883494,
+                              pvalue=0.19922132490385214)
+
+        The p-value is 0.1992, indicating a nearly 20% chance of observing
+        such an extreme value of the test statistic under the null hypothesis.
+        This exceeds 5%, so we do not reject the null hypothesis in favor of
+        the alternative.
+    """
+
+    args = _alexandergovern_input_validation(args, nan_policy)
+
+    # The following formula numbers reference the equation described on
+    # page 92 by Alexander, Govern. Formulas 5, 6, and 7 describe other
+    # tests that serve as the basis for equation (8) but are not needed
+    # to perform the test.
+
+    # precalculate mean and length of each sample
+    lengths = np.asarray([ma.count(arg) if nan_policy == 'omit' else len(arg)
+                          for arg in args])
+    means = np.asarray([np.mean(arg) for arg in args])
+
+    # (1) determine standard error of the mean for each sample
+    standard_errors = [np.std(arg, ddof=1) / np.sqrt(length)
+                       for arg, length in zip(args, lengths)]
+
+    # (2) define a weight for each sample
+    weights = ((1 / np.square(standard_errors)) /
+               np.sum(1 / np.square(standard_errors)))
+
+    # (3) determine variance-weighted estimate of the common mean
+    var_w = np.sum(weights * means)
+
+    # (4) determine one-sample t statistic for each group
+    t_stats = (means - var_w)/standard_errors
+
+    # calculate parameters to be used in transformation
+    v = lengths - 1
+    a = v - .5
+    b = 48 * a**2
+    c = (a * np.log(1 + (t_stats ** 2)/v))**.5
+
+    # (8) perform a normalizing transformation on t statistic
+    z = (c + ((c**3 + 3*c)/b) -
+         ((4*c**7 + 33*c**5 + 240*c**3 + 855*c) /
+          (b**2*10 + 8*b*c**4 + 1000*b)))
+
+    # (9) calculate statistic
+    A = np.sum(np.square(z))
+
+    # "[the p value is determined from] central chi-square random deviates
+    # with n_i - 1 degrees of freedom". Alexander, Govern (94)
+    p = distributions.chi2.sf(A, len(args) - 1)
+    return AlexanderGovernResult(A, p)
+
+
+def _alexandergovern_input_validation(args, nan_policy):
+    if len(args) < 2:
+        raise TypeError(f"2 or more inputs required, got {len(args)}")
+
+    # input arrays are flattened
+    args = [np.ravel(np.asarray(arg, dtype=float)) for arg in args]
+
+    for i, arg in enumerate(args):
+        if np.size(arg) <= 1:
+            raise ValueError("Input sample size must be greater than one.")
+        if arg.ndim != 1:
+            raise ValueError("Input samples must be one-dimensional")
+        if True in np.isinf(arg):
+            raise ValueError("Input samples must be finite.")
+
+        contains_nan, nan_policy = _contains_nan(arg, nan_policy=nan_policy)
+        if contains_nan and nan_policy == 'omit':
+            args[i] = ma.masked_invalid(arg)
```

Comment:
If we were to do this, it should probably go closer to the top. I imagine you would want to create masked arrays from the outset rather than turning everything into an array and then converting it to a masked array.

I don't really like masked arrays (and NumPy doesn't seem to either).

Here's one example of why: overhead.
```
In [1]: import numpy as np
In [2]: x = np.random.rand(10000)
In [3]: %timeit x.mean()
8.72 µs ± 255 ns per loop (mean ± std. dev. of 7 runs, 100000 loops each)
In [5]: y = np.ma.array(x)
In [6]: %timeit y.mean()
64.5 µs ± 1.46 µs per loop (mean ± std. dev. of 7 runs, 10000 loops each)
``` 
It is usually faster to just do
```
x = x[~np.isnan(x)]
```
and just proceed with it as a regular NumPy array.

But please hold off on addressing this.